stellargraph/connector/neo4j/mapper.py

```diff
@@ -289,3 +293,82 @@ def sample_features(self, head_nodes, batch_num):
             )
 
         return features
+
+
+@experimental(reason="the class is not fully tested")
+class Neo4jClusterNodeGenerator(ClusterNodeGenerator):
+    def __init__(self, graph, clusters=1, q=1, lam=0.1, name=None):
```

Comment:
already mentioned, but I think this init can basically call the superclass, though that will require a few changes to ClusterNodeGenerator, and some more methods (eg  check_graph_for_ml() ) and possibly abstract base for Neo4jStellarGraph.

Reply:
At the moment `ClusterNodGenerator.__init__` does some undesirable things when it comes to dealing with large graphs:

- duplicates and stores all node features in memory
- duplicates and stores all node ids in memory

The second of these `Neo4jClusterNodeGenerator` also does but in the future I'd like to have have "node cluster labels" stored in the Neo4j database. So i'd like to avoid (at least for now!) calling `ClusterNodeGenerator.__init__`